gnosis/protocol/gnosis_protocol_api.py

```diff
@@ -64,7 +64,25 @@ def __init__(self, ethereum_network: EthereumNetwork):
             self.network
         ]
         self.base_url = self.API_BASE_URLS[self.network]
-        self.http_session = requests.Session()
+        self.http_session = self._prepare_http_session()
+        self.request_timeout = request_timeout
+
+    def _prepare_http_session(self) -> requests.Session:
+        """
+        Prepare http session with custom pooling. See:
+        https://urllib3.readthedocs.io/en/stable/advanced-usage.html
+        https://2.python-requests.org/en/latest/api/#requests.adapters.HTTPAdapter
```

Comment:
This link doesn't open for me due SSL issues, I think that this works better https://docs.python-requests.org/en/v1.2.3/api/#requests.adapters.HTTPAdapter